scripts/zones/Batallia_Downs/Zone.lua

```diff
@@ -28,27 +46,34 @@ function onZoneIn( player, prevZone)
 
     if (triggerLightCutscene(player)) then -- Quest: I Can Hear A Rainbow
         cs = 901;
-    elseif (player:getCurrentMission(WINDURST) == tpz.mission.id.windurst.VAIN and player:getCharVar("MissionStatus") ==1) then
+    elseif (player:getCurrentMission(WINDURST) == tpz.mission.id.windurst.VAIN and player:getCharVar("MissionStatus") == 1) then
         cs = 903;
     end
 
+    if player:getCharVar("[QUEST]FullSpeedAhead") == 1 then
+        player:addStatusEffect(tpz.effect.FULL_SPEED_AHEAD, 0, 2, tpz.fsa.duration)
+    end
```

Comment:
Should this be first thing in this function and include a `return`? So that the other cutscenes don't fire when you zone in for this quest.